Cython/Compiler/Parsing.py

```diff
@@ -744,20 +744,25 @@ def p_atom_string(s):
         s.error("invalid string kind '%s'" % kind)
 
 
-def p_atom_ident_constants(s):
+def p_atom_ident_constants(s, bools_are_pybool=False):
     """
     Returns None if it isn't a special-cased named constant.
     Only calls s.next() if it successfully matches a named constant.
     """
     # s.sy == 'IDENT'
     pos = s.position()
     name = s.systring
+    result = None
+    if bools_are_pybool:
```

Comment:
On a second look I'm unconvinced that this is useful. Why it's needed is:

```
def xxx(x: cython.int):
    return x is True
```

returns a comparison `x==1`.

A bunch of the tests are like

```
       x = 0
        y = None
        match x:
            case False:
                y = 0
```

where the type of `x` gets inferred as C `int`, and thus the `is` comparison turns into `==0` rather than failing because it's not exactly `False`.

What I have here works, but maybe feels like the wrong solution. I'll have another look.